substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/logging/JfrLogConfiguration.java

```diff
@@ -0,0 +1,161 @@
+                if ((selection.wildcard && tagSet.getTags().containsAll(selection.tags))
+                        || (selection.tags.equals(tagSet.getTags()))) {
+                    level = Optional.of(selection.level);
+                    selection.matchesATagSet = true;
+                }
+            }
+            tagSet.setLevel(level);
+        }
+    }
+
+    private void verifySelections() {
+        for (JfrLogSelection selection : selections) {
+            if (!selection.matchesATagSet) {
+                Log.log().string("error: No tag set matches tag combination ")
+                    .string(selection.tags.toString().toLowerCase()).string(selection.wildcard ? "*" : "")
+                    .string(" for FlightRecorderLogging").newline();
+                System.exit(1);
+            }
+        }
+    }
+
+    private class JfrLogSelection {
+        private Set<JfrLogTag> tags = EnumSet.noneOf(JfrLogTag.class);
+        private LogLevel level = LogLevel.INFO;
+        private boolean wildcard = false;
+        private boolean matchesATagSet = false;
+
+        private void parse(String str) {
+            int equalsIndex;
+            if ((equalsIndex = str.indexOf('=')) > 0) {
+                try {
+                    level = LogLevel.valueOf(str.substring(equalsIndex + 1).toUpperCase());
+                } catch (IllegalArgumentException | NullPointerException e) {
+                    Log.log().string("error: Invalid log level '").string(str.substring(equalsIndex + 1))
+                        .string("' for FlightRecorderLogging. Use -XX:FlightRecorderLogging=help to see help.")
+                        .newline();
+                    System.exit(1);
+                }
+                str = str.substring(0, equalsIndex);
+            }
+
+            if (str.toUpperCase().equals("ALL")) {
+                wildcard = true;
+                return;
+            }
+
+            if (str.endsWith("*")) {
+                wildcard = true;
+                str = str.substring(0, str.length() - 1);
+            }
+
+            for (String s : str.split("\\+")) {
+                try {
+                    tags.add(JfrLogTag.valueOf(s.toUpperCase()));
+                } catch (IllegalArgumentException | NullPointerException e) {
+                    Log.log().string("error: Invalid log tag '").string(s)
+                        .string("' for FlightRecorderLogging. Use -XX:FlightRecorderLogging=help to see help.")
+                        .newline();
+                    System.exit(1);
+                }
+            }
+        }
+    }
+
+    private void printHelp() {
+        Log log = Log.log();
+        log.string("Usage: -XX:FlightRecorderLogging=[tag1[+tag2...][*][=level][,...]]").newline();
+        log.string("The syntax and behavior of this option is similar to that of the JDK's -Xlog option.").newline();
+        log.string("When this option is not set, logging is disabled.").newline();
```

Comment:
Is this the same on HotSpot? I think JFR logging is enabled there by default.

Reply:
Ah, yes it should be all `WARNING` by default. And there should also be an `-XX:FlightRecorderLogging=disable` option to disable all logging.

Reply:
There is also a thing where just having `-Xlog` sets everything to log at `info` level. I initially didn't implement this, since I thought it would look strange having `-XX:FlightRecorderLogging=` enable logging (whereas `-Xlog` looks more reasonable), and also some weirdness with the actual `RuntimeOptionKey`'s `defaultValue`. But I think it makes more sense to keep everything consistent, so I've also added this behaviour.